lib/quicly.c

```diff
@@ -3025,6 +3025,9 @@ static int _do_allocate_frame(quicly_conn_t *conn, quicly_send_context_t *s, siz
         /* adjust send_window to either 0 byte or MIN*2 bytes, if it is between those two */
         if (s->send_window < MIN_SEND_WINDOW * 2)
             s->send_window = s->send_window < MIN_SEND_WINDOW ? 0 : MIN_SEND_WINDOW * 2;
+        /* appropriate byte counting is applied only if the first frame for a datagram is ack-eliciting; we are too lazy to adjust
+         * things when a packet is turned into ack-eliciting after an ACK frame is written into the packet image. This diversion is
+         * considered acceptable as only the first packet being built would start with a non-ack-eliciting frame (i.e. ACK). */
```

Comment:
I agree with the sentiment of the comment, but I don't understand how this is appropriate byte counting. What precisely do you mean by it here?

Reply:
Glad you asked. The intent of the comment is to describe what I thought was the behavior. The behavior that I have assumed is as follows.

When the send window is below MTU, and quicly tries to allocate a new packet build buffer, the size of the buffer is set to:
* the size of the send window (with rounding) if the first frame to be written to the buffer is ack-eliciting, or
* MTU if the first frame to be written to the buffer is an ACK.

The latter is a deviation from appropriate byte counting.

However, the code does not actually work this way. The size of the buffer is set to MTU regardless of the type of the first frame.

I think we might want to discuss what the desired behavior is. @janaiyengar Thoughts?